src/resources/favorites.ts

```diff
@@ -0,0 +1,160 @@
+import { EthereumAddress, RainbowToken } from '@/entities';
+import { getUniswapV2Tokens } from '@/handlers/dispersion';
+import { createQueryKey, queryClient } from '@/react-query';
+import {
+  DAI_ADDRESS,
+  ETH_ADDRESS,
+  SOCKS_ADDRESS,
+  WBTC_ADDRESS,
+  WETH_ADDRESS,
+} from '@/references';
+import { useQuery } from '@tanstack/react-query';
+import { without } from 'lodash';
+
+export const favoritesQueryKey = createQueryKey(
+  'favorites',
+  {},
+  { persisterVersion: 1 }
+);
+
+const DEFAULT: Record<EthereumAddress, RainbowToken> = {
+  [DAI_ADDRESS]: {
+    address: DAI_ADDRESS,
+    color: '#F0B340',
+    decimals: 18,
+    favorite: true,
+    highLiquidity: true,
+    isRainbowCurated: true,
+    isVerified: true,
+    name: 'Dai',
+    symbol: 'DAI',
+    type: 'token',
+    uniqueId: DAI_ADDRESS,
+  },
+  [ETH_ADDRESS]: {
+    address: ETH_ADDRESS,
+    color: '#25292E',
+    decimals: 18,
+    favorite: true,
+    highLiquidity: true,
+    isVerified: true,
+    name: 'Ethereum',
+    symbol: 'ETH',
+    type: 'token',
+    uniqueId: ETH_ADDRESS,
+  },
+  [SOCKS_ADDRESS]: {
+    address: SOCKS_ADDRESS,
+    color: '#E15EE5',
+    decimals: 18,
+    favorite: true,
+    highLiquidity: true,
+    isRainbowCurated: true,
+    isVerified: true,
+    name: 'Unisocks',
+    symbol: 'SOCKS',
+    type: 'token',
+    uniqueId: SOCKS_ADDRESS,
+  },
+  [WBTC_ADDRESS]: {
+    address: WBTC_ADDRESS,
+    color: '#FF9900',
+    decimals: 8,
+    favorite: true,
+    highLiquidity: true,
+    isRainbowCurated: true,
+    isVerified: true,
+    name: 'Wrapped Bitcoin',
+    symbol: 'WBTC',
+    type: 'token',
+    uniqueId: WBTC_ADDRESS,
+  },
+};
+
+/**
+ * Returns a map of the given `addresses` to their corresponding `RainbowToken` metadata.
+ */
+async function fetchMetadata(addresses: string[]) {
+  const favoritesMetadata: Record<EthereumAddress, RainbowToken> = {};
+  const newFavoritesMeta = await getUniswapV2Tokens(
```

Comment:
note in scope here but we should prob replace this util with an updated one since its using a legacy endpoint, worth making a ticket

tagging @welps for any extra info

Reply:
https://linear.app/rainbow/issue/APP-854/replace-legacy-favorites-metadata-endpoint

Reply:
added context in ticket above